routing/pathfind.go

```diff
@@ -820,19 +842,21 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 				lnwire.TLVOnionPayloadOptional,
 			)
 
-			hop := route.Hop{
-				AmtToForward: amountToSend,
-				OutgoingTimeLock: uint32(
-					toNodeDist.incomingCltv,
-				),
-				LegacyPayload: !supportsTlv,
-			}
+			if edge.hopPayloadSize != nil {
```

Comment:
should we default to the normal clear hopsize calculation here or skipping the edge with an error log  is the way to go? I am undecided.

Reply:
I think that it's reasonable to default to cleartext hop size function? 

iiuc, this value will be non-nil for route hints / blinded paths and nil for regular hops that we got from the graph?

Reply:
hmm not quite, I am populating the value for non-blinded (normal) hops with the default: https://github.com/lightningnetwork/lnd/pull/7852/files#diff-d96272e3036db74758441649395efb5b51f96d67b1979c13a19a66c8c6ac9b1bR87-R96

That's why I tend to skip the edge because it should in the best case never happen. But also happen to use the default here as well.

Reply:
Prefer logging an error and skip to catch the unexpected case. Think we can just do
```go
if edge.hopPayloadSize == nil {
    log.Errorf("skipping edge=%v unable t...
    return
}

// now call the payload size fn.
```